File: services/app/apps/runner/dockers/python/runner.py
from io import StringIO  # Python 3
import sys
import json


class Runner:

    def __init__(self, args_list):
        self.args_list = args_list

    def call(self):
        tmp_stdout = StringIO()
        execution_results = []
        sys.stdout = tmp_stdout
        try:
            from solution import solution

            for args in self.args_list:
                try:
                    execution_results.append(json.dumps({
                        'type': 'result',
                        'value': solution(*args),
                        'output': tmp_stdout.getvalue()
                    }))
                except Exception as exc:
                    execution_results.append(json.dumps({
                        'type': 'error',
                        'value': exc.args,
                        'output': tmp_stdout.getvalue()
                    }))
                finally:
                    tmp_stdout.close()
                    tmp_stdout = StringIO()
                    sys.stdout = tmp_stdout
        except Exception as exc:
            execution_results.append(json.dumps({
                'type': 'error',
                'value': exc.args,
            }))

        sys.stdout = sys.__stdout__
        print('\n'.join(execution_results))

File: services/app/apps/runner/dockers/rust/runner.rs
// include!("solution");
//
//fn call(args_list)
//for each args_list call solutiona
//and write result into object
//return result list of results
    // error result JSON.dump(type: '', value: e)
    // normal_result
    /* JSON.dump(
      time: time,
      value: value,
      output: $stdout.string
    ) */
values = [[0, 1], [1,1], [1,0]]

global_result = []
foreach values, fn args do
try
output = capture_output
  timer.start
    restul  = solution(*args)
  timer.end

  (type: "result", time: timer.duration, value: restul, output: output)
 global_result  << restul
catch error
  (type: "error", value: error)
 global_result  << error_result
end
end
